apps/src/blockly/addons/functionEditor.js

```diff
@@ -370,4 +372,33 @@ export default class FunctionEditor {
     this.editorWorkspace.clear();
     Blockly.Events.enable();
   }
+
+  // If the definition block has a description property
+  // under definitionBlock.description, move it to the description field
+  // of the block, and delete the description property from the definition block.
+  // If the block does not have a description field, don't move the description.
+  moveDescriptionToBlockField(definitionBlock) {
+    if (
+      definitionBlock.description &&
+      definitionBlock.inputList &&
+      definitionBlock.inputList[1]?.fieldRow &&
+      definitionBlock.inputList[1].fieldRow[1]
+    ) {
+      definitionBlock.inputList[1].fieldRow[1].setValue(
+        definitionBlock.description
+      );
+      definitionBlock.description = undefined;
```

Comment:
Is there a difference between deleting the key and setting it to undefined? Should we delete it?

Reply:
good call, I'll delete it. For our purposes I don't think there's much of a difference, but might as well delete.